core/dbt/main.py

```diff
@@ -191,14 +191,15 @@ def handle_and_check(args):
         if parsed.record_timing_info:
             profiler_enabled = True
 
-        with profiler(enable=profiler_enabled, outfile=parsed.record_timing_info):
+        if profiler_enabled:
```

Comment:
This logic right now would leave current dbt not running `run_from_args` if profiler is not enabled

Reply:
Closed the PR to work on feature/click-cli branch instead of main